build-system/common/esbuild-babel.js

```diff
@@ -48,27 +109,30 @@ function getEsbuildBabelPlugin(
   preSetup = () => {},
   postLoad = () => {}
 ) {
-  function sha256(contents) {
+  function sha256(obj) {
     if (!enableCache) {
       return '';
     }
-    const hash = crypto.createHash('sha256');
-    hash.update(callerName);
-    hash.update(contents);
-    return hash.digest('hex');
+    return crypto
+      .createHash('sha256')
+      .update(JSON.stringify(obj))
+      .digest('hex');
   }
 
-  function batchedRead(path) {
+  /**
+   * @param {string} path
+   * @param {string} optionsHash
+   * @returns {{contents: string, hash: string}}
+   */
+  function batchedRead(path, optionsHash) {
     let read = readCache.get(path);
     if (!read) {
       read = fs.promises
-        .readFile(path)
-        .then((contents) => {
-          return {
-            contents,
-            hash: sha256(contents),
-          };
-        })
+        .readFile(path, 'utf8')
```

Comment:
Thanks for the tip! Done

Reply:
Unresolving this to point out an interesting discovery: hashing goes way faster for utf8 string than for `Buffer`. I don't know why. This change along made "Time spent hashing" go from 20ms --> 120ms.

Also, I saw that `babel-loader` is using md5 instead of sha256 because it is slightly faster. Any objection to me switching?